src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/readers/RecursiveJSONParser.java

```diff
@@ -127,6 +131,26 @@ private ObjectNode expandIncludesInObject(ObjectNode input) throws IOException {
             result = processMergeDirective(result);
         }
 
+        // iterate here again to put DYNAMIC_INCLUDE_FLAT_KEY as a parent node
+        if (result.has(INCLUDE_FLAT_KEY)) {
+            ObjectNode dynamicIncludeFlatResult = mapper.getNodeFactory().objectNode();
+            Iterator<String> namesSecond = input.fieldNames();
+            while (namesSecond.hasNext()) {
+                String name = namesSecond.next();
+                JsonNode node = input.get(name);
+                if (isDynamicIncludeFlat(node)) {
+                    String key = DYNAMIC_INCLUDE_FLAT_KEY.concat(name);
```

Comment:
doesn't have that much sense this here. It was done for the dynamic merge because in that case we could have different attribute names with the merge directive and by replacing all of them with a unique key in the same node we would have come to repeat attribute names. In this case there is no way to avoid this since we would always have the $includeFlat key.

Reply:
done